src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -437,11 +437,20 @@ export class StudyViewPageStore {
             cols: cols,
             grid: STUDY_VIEW_CONFIG.layout.grid,
             gridMargin: STUDY_VIEW_CONFIG.layout.gridMargin,
-            layout: calculateLayout(this.visibleAttributes, cols),
+            layout: calculateLayout(this.visibleAttributes, cols, this.currentGridLayout, this.currentGridElement),
             dimensions: STUDY_VIEW_CONFIG.layout.dimensions
         };
     }
 
+    @autobind
+    @action
+    updateCurrentGridLayout(currentGridLayout: ReactGridLayout.Layout[]) {
+        this.currentGridLayout = currentGridLayout;
+    }
+
+    private currentGridLayout: ReactGridLayout.Layout[] | undefined = undefined;
+    private currentGridElement: ChartMeta | undefined = undefined;
```

Comment:
This is great. Now you have a clear picture how this `currentGridElement` is used. Let's name this variable more clearly. `currentFocusedChartByUser`? We don't use `element` concept in anywhere yet, so let's avoid this.